hail/src/main/scala/is/hail/expr/types/BaseType.scala

```diff
@@ -25,14 +25,17 @@ trait Requiredness {
 }
 
 object BaseStruct {
-  def getMissingness[Ty <: BaseType with Requiredness](types: Array[Ty], missingIdx: Array[Int]): Int = {
-    assert(missingIdx.length == types.length)
+  def getSetMissingness(req: Array[Boolean], missingIdx: Array[Int]): Int = {
```

Comment:
Agreed with the name change. I think this could be clearer if getMissingness returned the array of missing values alongside the final count. I think it would be clearer than having the construction of the missingness array split between the caller and getSetMissingness.

```scala
 def getSetMissingness(req: Array[Boolean]): (Array[Int], Int) = {
   var missingIdx = ArrayBuilder.ofInt
   
   # ... same 

   (missingIdx, i)
```